scipy/optimize/_highs/pyHiGHS/src/highs_wrapper.pyx

```diff
@@ -0,0 +1,622 @@
+                and `parallel=True` then PAMI will be used.
+            - time_limit : double
+                Max number of seconds to run the solver for.
+            - use_original_HFactor_logic : bool
+                Undocumented advanced option.
+            - write_solution_to_file : bool
+                Write the primal and dual solution to a file
+            - write_solution_pretty : bool
+                Write the primal and dual solution in a pretty
+                (human-readable) format
+
+        See [2]_ for a list of all non-advanced options.
+
+    Returns
+    -------
+    res : dict
+
+        If model_status is one of OPTIMAL,
+        REACHED_DUAL_OBJECTIVE_VALUE_UPPER_BOUND, REACHED_TIME_LIMIT,
+        REACHED_ITERATION_LIMIT:
+
+            - `status` : int
+                Model status code.
+            - `message` : str
+                Message corresponding to model status code.
+            - `x` : list
+                Solution variables.
+            - `slack` : list
+                Slack variables.
+            - `lambda` : list
+                Lagrange multipliers assoicated with the constraints
+                Ax = b.
+            - `s` : list
+                Lagrange multipliers associated with the constraints
+                x >= 0.
+            - `fun`
+                Final objective value.
+            - `simplex_nit` : int
+                Number of iterations accomplished by the simplex
+                solver.
+            - `ipm_nit` : int
+                Number of iterations accomplished by the interior-
+                point solver.
+
+        If model_status is not one of the above:
+
+            - `status` : int
+                Model status code.
+            - `message` : str
+                Message corresponding to model status code.
+
+    Notes
+    -----
+    If `options['write_solution_to_file']` is `True` but
+    `options['solution_file']` is unset or `''`, then the solution
+    will be printed to `stdout`.
+
+    If any iteration limit is reached, no solution will be
+    available.
+
+    `OptimizeWarning` will be raised if any option value set by
+    the user is found to be incorrect.  Some invalud values for
+    `message_level` are known to fail to raise these warnings
+    (3, 5, and 6 to be exact).
+
+    References
+    ----------
+    .. [1] https://www.maths.ed.ac.uk/hall/HiGHS
+    .. [2] https://www.maths.ed.ac.uk/hall/HiGHS/HighsOptions.html
+    '''
+
+
+    cdef int numcol = c.size
+    cdef int numrow = rhs.size
+    cdef int numnz = avalue.size
+
+    # Fill up a HighsLp object
+    cdef HighsLp lp
+    lp.numCol_ = numcol
+    lp.numRow_ = numrow
+
+    lp.colCost_.resize(numcol)
+    lp.colLower_.resize(numcol)
+    lp.colUpper_.resize(numcol)
+
+    lp.rowLower_.resize(numrow)
+    lp.rowUpper_.resize(numrow)
+    lp.Astart_.resize(numcol + 1)
+    lp.Aindex_.resize(numnz)
+    lp.Avalue_.resize(numnz)
+
+    # Be careful not index into nothing
```

Comment:
this comment is a little cryptic - expand?

Reply:
Expanded